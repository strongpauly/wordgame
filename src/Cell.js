import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Cell extends Component {

  static propTypes = {
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
    char: PropTypes.string.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      selected: false
    };

    this.selectCell = this.selectCell.bind(this);
  }

  selectCell () {
    this.setState({selected: !this.state.selected});
  }

  render() {
    let className = ['cell'];
    let content = this.props.char;
    if (this.state.selected) {
      className.push('selected');
    }
    return <td className={ className.join(' ') } onClick={this.selectCell}>{ content }</td>;
  }
}
