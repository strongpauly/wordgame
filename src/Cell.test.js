import React from 'react';
import Cell from './Cell';
import {mount, shallow} from 'enzyme';

/* eslint-env jest */

describe('<Cell>', () => {

  it('renders without crashing', () => {
    shallow(<Cell x={0} y={0} char="C"/>);
  });

  it('has "cell" css class', () => {
    const cell = shallow(<Cell x={0} y={0} char="C"/>);
    expect(cell).toMatchSnapshot();
    expect(cell.find('.cell')).toHaveLength(1);
  });

  it('click adds "selected" css class', () => {
    let table = mount(<table><tbody><tr><Cell x={0} y={0} char="C"/></tr></tbody></table>);
    expect(table.find('.selected')).toHaveLength(0);
    table.find('td').simulate('click');
    expect(table.find('.selected')).toHaveLength(1);
  });

});
