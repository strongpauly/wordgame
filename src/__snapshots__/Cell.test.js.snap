// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell> has "cell" css class 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <td
    className="cell"
    onClick={[Function]}
>
    C
</td>,
  "nodes": Array [
    <td
      className="cell"
      onClick={[Function]}
>
      C
</td>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Cell
        char="C"
        x={0}
        y={0}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Cell {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "char": "C",
          "x": 0,
          "y": 0,
        },
        "refs": Object {},
        "selectCell": [Function],
        "state": Object {
          "selected": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <td
          className="cell"
          onClick={[Function]}
>
          C
</td>,
        "_debugID": 4,
        "_renderedOutput": <td
          className="cell"
          onClick={[Function]}
>
          C
</td>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Cell
    char="C"
    x={0}
    y={0}
/>,
}
`;
